src/snapred/backend/data/LocalDataService.py

```diff
@@ -103,70 +100,33 @@ def _readInstrumentParameters(self) -> Dict[str, Any]:
 
     def readStateConfig(self, runId: str) -> StateConfig:
         reductionParameters = self._readReductionParameters(runId)
+        diffCalRecord = self.readCalibrationRecord(runId)
+        if diffCalRecord is None:
+            diffCalRecord = self.readCalibrationState(runId)
+            particleBounds = diffCalRecord.instrumentState.particleBounds
+        else:
+            diffCalRecord = diffCalRecord.calibrationFittingIngredients
+            particleBounds = diffCalRecord.calibrationFittingIngredients.instrumentState.particleBounds
```

Comment:
Can you call the first instance of `diffCalRecord` something else and set its type, like
``` python
previousDiffCalRecord: CalibrationRecord = self.readCalibrationRecord(runId)
if previousDiffCalRecord is None:
    diffCalibration: Calibration = self.readCalibrationRecord(runId)
else:
    diffCalibration: Calibration = previousDiffCalRecord.calibrationFittingIngredients
particleBounds = diffCalibration.instrumentState.particleBounds
```

Reply:
Also, I think L109 as it's written is an error.  L108 sets `diffCalRecord` to be a `Calibration` object, and `Calibration` objects do not have a `calibrationFittingIngredients` property.  The fix i suggested should fix that.